src/cim/components/featureSupportLevels/ACMFeatureSupportLevelProvider.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { PropsWithChildren } from 'react';
+
+import { getOCPVersions } from '../helpers';
+import {
+  FeatureSupportLevelsMap,
+  FeatureIdToSupportLevel,
+  FeatureId,
+  SupportLevel,
+  FeatureSupportLevelContextProvider,
+  FeatureSupportLevelData,
+} from '../../../common';
+import { ClusterImageSetK8sResource } from '../../types';
+
+export type ACMFeatureSupportLevelProvider = PropsWithChildren<{
+  clusterImages: ClusterImageSetK8sResource[];
+}>;
+
+export const ACMFeatureSupportLevelProvider: React.FC<ACMFeatureSupportLevelProvider> = ({
+  children,
+  clusterImages,
+}) => {
+  const ocpVersions = React.useMemo(() => getOCPVersions(clusterImages), [clusterImages]);
+
+  const supportLevelData: FeatureSupportLevelsMap = React.useMemo<FeatureSupportLevelsMap>(() => {
+    const featureSupportLevelsMap: FeatureSupportLevelsMap = {};
+
+    ocpVersions.forEach((version) => {
+      // TODO(mlibra): get this dynamically
+      const featureIdToSupportLevel: FeatureIdToSupportLevel = {
+        REQUESTED_HOSTNAME: 'supported',
+        SNO: 'supported',
+        CNV: 'supported',
+        NETWORK_TYPE_SELECTION: 'supported',
+      };
+      featureSupportLevelsMap[version.version] = featureIdToSupportLevel;
+    });
+    return featureSupportLevelsMap;
+  }, [ocpVersions]);
+
+  const getVersionSupportLevelsMap: FeatureSupportLevelData['getVersionSupportLevelsMap'] = React.useCallback(
```

Comment:
you can simply return undefined here

Reply:
Right. I could simplify whole component but I consider it as a skeleton, expected to be changed once we properly retrieve the support map.

Reply:
makes sense